From b19384d3c130858bb31a343ea8fce26be71b5998 Mon Sep 17 00:00:00 2001
From: Michael Hudson-Doyle <michael.hudson@canonical.com>
Date: Thu, 20 Sep 2018 09:20:31 +0200
Subject: [PATCH] arm64/arm64asm: recognise new ssbb/pssbb mnemonics from
 objdump

Fixes golang/go#27754

Change-Id: I8fcc3bc3c718cf0d93afbd1d383df48316b522d4
Reviewed-on: https://go-review.googlesource.com/136455
Run-TryBot: Michael Hudson-Doyle <michael.hudson@canonical.com>
TryBot-Result: Gobot Gobot <gobot@golang.org>
Reviewed-by: Cherry Zhang <cherryyz@google.com>
---
 arm64/arm64asm/objdump_test.go | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/src/cmd/vendor/golang.org/x/arch/arm64/arm64asm/objdump_test.go
+++ b/src/cmd/vendor/golang.org/x/arch/arm64/arm64asm/objdump_test.go
@@ -98,6 +98,13 @@
 			return true
 		}
 	}
+	// New objdump supports some newer mnemonics than this package. This
+	// package should be updated to support the new mnemonics and the sense
+	// of this reversed to continue passing with older objdumps but that
+	// requires internal ARM tooling.
+	if newForm, ok := newMnemonics[text]; ok && newForm == dec.text {
+		return true
+	}
 	// GNU objdump misses spaces between operands for some instructions (e.g., "ld1 {v10.2s, v11.2s}, [x23],#16")
 	if strings.Replace(text, " ", "", -1) == strings.Replace(dec.text, " ", "", -1) {
 		return true
@@ -143,3 +150,8 @@
 	"strh":  "sturh",
 	"prfm":  "prfum",
 }
+
+var newMnemonics = map[string]string{
+	"dsb #0x00": "ssbb",
+	"dsb #0x04": "pssbb",
+}
